### PR TITLE
Simplify key codes expression in Tags component

### PR DESCRIPTION
**Pull request by stesel**
## What type of PR is this? (check all applicable)

- [x] Refactor

## Description
Move combination key codes to separate const in `tags.jsx` file.
Add test for checking `preventDefault` in `handleKeyDown` method.

Suggestions comes form codeclimate.
https://codeclimate.com/github/thepracticaldev/dev.to/app/javascript/shared/components/tags.jsx/source#issue-a087ee1bf368783e621a0b4563a3aaeb

## Related Tickets & Documents
https://codeclimate.com/github/thepracticaldev/dev.to/app/javascript/shared/components/tags.jsx/source#issue-a087ee1bf368783e621a0b4563a3aaeb

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: app/javascript/shared/components/tags.jsx [hunk 1]
```diff
@@ -289,14 +297,7 @@ class Tags extends Component {
...
       }
-    } else if (
-      (e.keyCode < 65 || e.keyCode > 90) &&
-      e.keyCode != KEYS.COMMA &&
-      e.keyCode != KEYS.DELETE &&
-      e.keyCode != KEYS.LEFT &&
-      e.keyCode != KEYS.RIGHT &&
-      e.keyCode != KEYS.TAB
-    ) {
+    } else if ((e.keyCode < 65 || e.keyCode > 90) && !COMBO_KEYS.includes(e.keyCode)) {
```
Comment (rhymes) [inline app/javascript/shared/components/tags.jsx]: we should probably name 65 and 90's too

Comment (jacobherrington) [inline app/javascript/shared/components/tags.jsx]: Could we line wrap this conditional? It's just a bit long for reading on one line in my opinion. 😬

Comment (stesel) [reply app/javascript/shared/components/tags.jsx]: Sure, easy :)

Comment (stesel) [reply app/javascript/shared/components/tags.jsx]: After migration to event.key I'd suggest to use RegExp for letters range.

**Comment by stesel** [comment]
> Hi @stesel thanks for the code! The PR looks nice.
> 
> A question: would you be willing to go a step further and switch from `event.keyCode`, which is deprecated, to use `event.key`, see https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key

Sure, I've migrated to event.key.

Now I have questions, do we need fallbacks for old browsers witch doesn't support `KeyboardEvent.key` well?

**Comment by rhymes** [comment]
@stesel I don't think so, I'm looking at both https://caniuse.com/#feat=keyboardevent-key and https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent#Browser_compatibility and I think we're fine. 

It's obviously open to discussion...

**Comment by stesel** [comment]
@rhymesfor thank you for response!
Then PR can be reviewed again.

**Pull request merged**